src/Aspire.Hosting.NodeJs/NodeExtensions.cs

```diff
@@ -57,6 +57,58 @@ public static IResourceBuilder<NodeAppResource> AddNpmApp(this IDistributedAppli
                       .WithArgs(allArgs);
     }
 
+    /// <summary>
+    /// Adds a node application to the application model. Executes the command with the specified script name.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="nodeExecutor">The executor to use. Can be a package manager like npm, pnpm, yarn or directly node.</param>
+    /// <param name="scriptName">The script to execute.</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    private static IResourceBuilder<NodeAppResource> AddGenericNodeApp(this IDistributedApplicationBuilder builder, string name, string workingDirectory, string nodeExecutor, string scriptName, string[]? args = null)
```

Comment:
todo: we should separate private and public methods to make the intents clear (maybe put the generic private method at the end of the class)